lib/Utils/Architectures.pm

```diff
@@ -126,4 +128,30 @@ sub is_ppc64le {
     return check_var('ARCH', 'ppc64le');
 }
 
+=head2 is_orthos_machine
+
+ is_orthos_machine();
+
+Returns C<true if machine FQDN has arch.suse.de suffix>.
+
+=cut
+sub is_orthos_machine {
+    my $sut_fqdn = get_var('SUT_IP', 'nosutip');
+    return 1 if $sut_fqdn =~ /(arch\.suse\.de)/im;
+    return 0;
+}
+
+=head2 is_covered_machine
+
+ is_covered_machine();
+
+Returns C<true if machine FQDN has qa.suse.de, qa2.suse.asia or arch.suse.de suffix>.
+
+=cut
+sub is_covered_machine {
+    my $sut_fqdn = get_var('SUT_IP', 'nosutip');
+    return 1 if $sut_fqdn =~ /(arch\.suse\.de|qa2\.suse\.asia|qa\.suse\.de)/im;
```

Comment:
what if SUT_IP is direct ip without fqdn? For worker setting, SUT_IP does not have any requirement that this value needs to be fqdn.

Reply:
You can choose to use it as fqdn or not. That is easy.

Reply:
The problem is if it is configured by ip, this new code logic will break the test on it because it considers it as not covered machine and goes to other branches which is wrong.

Reply:
1. So you just fqdn, that will be fine
2. Most SUT_IP in salt pillar settings use fqdn
3. Only very few ipmi aarch64 machines use fqdn

Reply:
It affects many users' configuration rule for SUT_IP. If this PR is merged, the new restriction by it to use fqdn should become forceful now. What's your plan to broadcast it? I am willing to merge it if this can be well handled.

Reply:
Salt pillar worker settings only have 7 ppc64les that use ip address, so fqdn has already been chosen by the most instead of happen by being forced. I can help add comments or warnings to the top of workers setting file in salt pillar to make it more transparent, if you think this is very necessary.

Reply:
That will be perfect.